moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -231,16 +259,28 @@ class Robot
            const unsigned int num_steps);
 
   /**
-   * @brief interpolate - interpolates from \e from point towards \e to
+   * @brief interpolate - Cartesian interpolation from \e from point towards \e to
    * point at time t in [0,1] and stores it in \e point.
-   * @param from: initial point
-   * @param to: final point
+   * @param from: initial Cartesian point
+   * @param to: final Cartesian point
    * @param t: parameteric time
-   * @param point: interpolated point
+   * @param point: interpolated Cartesian point
    */
   void interpolate( const std::unique_ptr<CartTrajectoryPoint>& from,
-                         const std::unique_ptr<CartTrajectoryPoint>& to,
-                         double t, std::unique_ptr<TrajectoryPoint> & point);
+                    const std::unique_ptr<CartTrajectoryPoint>& to,
+                    double t, std::unique_ptr<CartTrajectoryPoint> & point);
+
+  /**
+   * @brief interpolate - Joint interpolation from \e from point towards \e to
+   * point at time t in [0,1] and stores it in \e point.
+   * @param from: initial Joint point
+   * @param to: final Joint point
+   * @param t: parameteric time
+   * @param point: interpolated Joint point
+   */
+  void interpolate( const std::unique_ptr<JointTrajectoryPoint>& from,
```

Comment:
This method doesn't change the underlying class.  It should have the `const` keyword appended to indicate this.